engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTextureFromFBO.java

```diff
@@ -136,36 +127,29 @@ private int fetchTextureId() {
                 case LightAccumulationTexture:
                     return inputFbo.lightBufferTextureId;
             }
-
         } else {
-            logger.warning("FBOs manager has no record of an FBO named " + this.fboURN.toString());
+            logger.warning("FBOs manager has no record of an FBO named " + this.fboUrn.toString());
         }
 
         return 0;
     }
 
     /**
-     * Normally called by the FBO manager provided on construction, when the FBOs are regenerated.
+     * Normally called by the FBO manager provided, when the FBOs are regenerated.
      *
      * This method refreshes the task's reference to the FBO attachment, so that they are always up to date.
      */
     @Override
     public void update() {
-
-        inputFbo = fbosManager.get(fboURN);
-
-        // If a node taking advantage of this state change is disabled when a game is started, task is null.
-        // This is due to the task list generator not having yet called the generateTask() method.
-        if (task != null) {
-            task.setTextureId(fetchTextureId());
-        }
+        inputFbo = fboManager.get(fboUrn);
+        task.setTextureId(fetchTextureId());
```

Comment:
Because, that was just an edge case that had to be handled because of badly designed code. As things stand, the update() function will only be called after subscription to fboManager, and the subscription fo fboManager takes place only after creation of task (in generateDefaultTask). As things stand, task can never be null inside update(), no matter what the conditions.